### ISSUE TITLE
consecutive_assertion_linter() gives false positive when interceded by '=' assignment

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2444]
```r
lint("
stopifnot(a)
b=c
stopifnot(d)
", consecutive_assertion_linter())
# <text>:2:1: warning: [consecutive_assertion_linter] Unify consecutive calls to stopifnot().
# stopifnot(a)
# ^~~~~~~~~~~~
```

Almost surely this is because of the weird `<expr_or_assign_or_help>` node that is usually `<expr>`.
